src/datepicker/jalali/ngb-calendar-jalali.ts

```diff
@@ -0,0 +1,79 @@
+/**
+ * Created by mehrabisajad on 2017/03/28.
+ */
+import {NgbDate} from '../ngb-date';
+import {NgbPeriod, NgbCalendar} from '../ngb-calendar';
+import {Injectable} from '@angular/core';
+import {isInteger} from '../../util/util';
+
+
+@Injectable()
+export abstract class NgbCalendarJalali extends NgbCalendar {
+  getDaysPerWeek() { return 7; }
+
+  getMonths() { return [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]; }
+
+  getWeeksPerMonth() { return 6; }
+
+  isValid(date: NgbDate): boolean {
+    return date && isInteger(date.year) && isInteger(date.month) && isInteger(date.day) &&
+        !isNaN(this.toGregorian(date).getTime());
+  }
+
+  setDay(date: NgbDate, day: number): NgbDate {
+    let mDays = this.getDaysInJalaliMonth(date.month, date.year);
+    if (day <= 0) {
+      while (day <= 0) {
+        date = this.setMonth(date, date.month - 1);
+        mDays = this.getDaysInJalaliMonth(date.month, date.year);
+        day += mDays;
+      }
+    } else if (day > mDays) {
+      while (day > mDays) {
+        day -= mDays;
+        date = this.setMonth(date, date.month + 1);
+        mDays = this.getDaysInJalaliMonth(date.month, date.year);
+      }
+    }
+    date.day = day;
+    return date;
+  }
+
+  setMonth(date: NgbDate, month: number): NgbDate {
+    month = +month;
+    date.year = date.year + Math.floor((month - 1) / 12);
+    date.month = Math.floor(((month - 1) % 12 + 12) % 12) + 1;
```

Comment:
i think it needs to be like this to calculate negative "month". although my reference to this code is from : 
/src/datepicker/hijri/ngb-calendar-hijri.ts#L39-L44